src/libponyrt/sched/scheduler.c

```diff
@@ -1179,11 +1179,15 @@ bool ponyint_sched_unmute_senders(pony_ctx_t* ctx, pony_actor_t* actor)
       if(!has_flag(to_unmute, FLAG_UNSCHEDULED))
       {
         ponyint_unmute_actor(to_unmute);
-        // TODO: we don't want to reschedule if our queue is empty.
-        // That's wasteful.
-        ponyint_sched_add(ctx, to_unmute);
-        DTRACE2(ACTOR_SCHEDULED, (uintptr_t)sched, (uintptr_t)to_unmute);
-        actors_rescheduled++;
+
+        // Only reschedule if the queue isn't empty. The queue hasn't been
+        // marked as empty when the actor was muted.
+        if(!ponyint_messageq_markempty(&to_unmute->q))
```

Comment:
i think tried this before and something bad happened, i don't remember what. i'd have to look at what markempty does because i think this was causing things to deadlock.

Reply:
@SeanTAllen Did you have any chance to look into this?

Reply:
Not yet. Work has been rather busy.